### DISCUSSION TITLE
RSS Feed

### DISCUSSION THREAD

**Discussion by ChristophLeuker** [neuefische/questions#631] [Web] [answered]
Hey,
i want to implement a News Feed on my Landing Page. 
I tried to use Parser from rss-parser but it didnt work correctly. 
This is my component: 
![rssfeed-code](https://github.com/neuefische/questions/assets/131157050/d6609c55-a0dd-4a1f-a5d9-9050d7e88d8a)
and when i try to render it, i get this message:
![rssfeed-fehler](https://github.com/neuefische/questions/assets/131157050/a846fc20-6a8e-4db6-88c7-a0cf3b298b40)

Any ideas about that? Or maybe an easier way to get a News Feed for Darts? 

This is the website i´m trying to use. : https://www.dartn.de/Webnapping

I pushed the newest branch on https://github.com/ChristophLeuker/Capstone-project-Throw-In-App

Thank you for help :)

**Comment by actyvystom** [comment, marked as answer]
Hey @ChristophLeuker,
please provide code snippets instead of screenshots (next time 😉 ).
You should **never** perform direct API calls to an external data source from within `Next.js` clients. Wrapping your rss request in an internal API route then also eliminates the CORS-policy issues and you should be able to parse the RSS response within your (internal) API handler using your `rss-parser` and return the parsed data to your client/page component.
Hope that helps!
Thomas

**Comment by ChristophLeuker** [comment]
Needed some ChatGPT help to build the code but i understand what you mean and so far, it looked good ;) Thanks !
